ush/ioda/bufr2ioda/bufr2ioda_adpupa_prepbufr.py

```diff
@@ -0,0 +1,444 @@
+    # Make the QuerySet for all the data we want
+    # ============================================
+    start_time = time.time()
+
+    logger.info('Making QuerySet')
+    q = bufr.QuerySet(subsets)
+
+    # MetaData
+    q.add('latitude', 					'*/PRSLEVEL/DRFTINFO/YDR')
+    q.add('longitude', 					'*/PRSLEVEL/DRFTINFO/XDR')
+    q.add('stationIdentification', 			'*/SID')
+    q.add('stationElevation', 				'*/ELV')
+    q.add('timeOffset', 				'*/PRSLEVEL/DRFTINFO/HRDR')
+    q.add('releaseTime', 				'*/PRSLEVEL/DRFTINFO/HRDR')
+    q.add('temperatureEventProgramCode',		'*/PRSLEVEL/T___INFO/T__EVENT{1}/TPC')
+    q.add('pressure', 					'*/PRSLEVEL/P___INFO/P__EVENT{1}/POB')
+
+    # ObsValue
+    q.add('verticalSignificance', 			'*/PRSLEVEL/CAT')
+    q.add('stationPressure', 				'*/PRSLEVEL/P___INFO/P__EVENT{1}/POB')
+    q.add('airTemperature', 				'*/PRSLEVEL/T___INFO/T__EVENT{1}/TOB')
+    q.add('virtualTemperature', 			'*/PRSLEVEL/T___INFO/TVO')
+    q.add('specificHumidity', 				'*/PRSLEVEL/Q___INFO/Q__EVENT{1}/QOB')
+    q.add('windEastward', 				'*/PRSLEVEL/W___INFO/W__EVENT{1}/UOB')
+    q.add('windNorthward', 				'*/PRSLEVEL/W___INFO/W__EVENT{1}/VOB')
+    q.add('height', 					'*/PRSLEVEL/Z___INFO/Z__EVENT{1}/ZOB')
+
+    # QualityMark
+    q.add('pressureQM', 				'*/PRSLEVEL/P___INFO/P__EVENT{1}/PQM')
+    q.add('airTemperatureQM', 				'*/PRSLEVEL/T___INFO/T__EVENT{1}/TQM')
+    q.add('virtualTemperatureQM', 			'*/PRSLEVEL/T___INFO/T__EVENT{1}/TQM')
+    q.add('specificHumidityQM', 			'*/PRSLEVEL/Q___INFO/Q__EVENT{1}/QQM')
+    q.add('windEastwardQM', 				'*/PRSLEVEL/W___INFO/W__EVENT{1}/WQM')
+    q.add('windNorthwardQM', 				'*/PRSLEVEL/W___INFO/W__EVENT{1}/WQM')
+
+    end_time = time.time()
+    running_time = end_time - start_time
+    logger.debug(f'Running time for making QuerySet : {running_time} seconds')
+
+    # ==============================================================
+    # Open the BUFR file and execute the QuerySet to get ResultSet
+    # Use the ResultSet returned to get numpy arrays of the data
+    # ==============================================================
+    start_time = time.time()
+
+    logger.info('Executing QuerySet to get ResultSet')
+    with bufr.File(DATA_PATH) as f:
+        r = f.execute(q)
+
+    logger.info('Executing QuerySet: get metadata')
+    # MetaData
+    lat = r.get('latitude', 				'verticalSignificance')
+    lon = r.get('longitude', 				'verticalSignificance')
+    lon[lon>180] -= 360  				# Convert Longitude from [0,360] to [-180,180]
+    sid = r.get('stationIdentification', 		'verticalSignificance')
+    elv = r.get('stationElevation', 			'verticalSignificance', type='float')
+    tpc = r.get('temperatureEventProgramCode', 		'verticalSignificance')
+    pob = r.get('pressure', 				'verticalSignificance')
+    pob *= 100
+
+    # Time variable
+    hrdr = r.get('timeOffset', 				'verticalSignificance')
+    ulan = r.get('releaseTime')
+    ulan = np.int64(ulan*3600)
```

Comment:
I'm guessing `ulan` is some BUFR/prepBUFR name, but perhaps we can use more descriptive variable names in this script

Reply:
@CoryMartin-NOAA _ulan_ is not even a BUFR/prepBUFR name. I used it because jscda is using the same.